ddd-core/src/main/kotlin/io/paju/ddd/AggregateRoot.kt

```diff
@@ -49,6 +50,31 @@ abstract class AggregateRoot<S: State, E: StateChangeEvent>
         }
     }
 
+    protected fun expectInitializedState() {
+        if(!isInitialized()) {
+            throw InvalidStateException("Instance not in expected state. " +
+                "Instance state was uninitialized, expected initialized", this)
+        }
+    }
+
+    protected fun expectUninitializedState() {
+        if(isInitialized()) {
+            throw InvalidStateException("Instance not in expected state. " +
+                "Instance state was initialized, expected uninitialized", this)
+        }
+    }
+
+    protected inline fun <reified T: State>expectState(): T {
```

Comment:
All of these are nice. A suggestion: one more extension that takes a lambda `T -> Boolean` to further pin down the state.

Although at that point it might be worth thinking about a full DSL, so might be overkill at this stage.

Reply:
Nice 👍  Now there is one lambda based check.